### PR TITLE
regex + new personality!

### PR DESCRIPTION
**Pull request by tobyab**

### CODE REVIEW

**Review by grymmy — CHANGES_REQUESTED** [review]
I think we also want to extract the puzzle ID with the regular expression as well.

File: app.py [hunk 1]
```diff
@@ -100,7 +105,15 @@ def on_new_tweet():
         return "weird!"
 
     puzzle_id = parts[0]
```
Comment (grymmy) [inline app.py]: See where the variables "parts" and "text_parts" are above - the goal is to replace those variables, and all of the text splitting and error checking, with a simple multivariable regular expression, doing the validation all in one swoop...

Comment (grymmy) [reply app.py]: See this Stack Overflow post for more info on how one can extract multiple named regex vars: https://stackoverflow.com/questions/1800817/how-can-i-get-part-of-regex-match-as-a-variable-in-python

The top answer has an example of two named vars within the regex and how they can be extracted.

Comment (tobyab) [reply app.py]: Hey @grymmy! Just added your suggested changes

**Review by grymmy — APPROVED** [review]

**Pull request merged**